lib/puppet/provider/archive/ruby.rb

```diff
@@ -219,6 +219,53 @@ def s3_download(path)
     aws(params)
   end
 
+  def check_latest
+    # Check if the file exists, just download it if not
+    if File.exist?(archive_filepath)
+      localmodifytime = File.mtime(archive_filepath).to_i
+      case resource[:source]
+      when %r{^(http)}
+        uri = URI(resource[:source])
+        response = nil
+        Net::HTTP.start(uri.host, uri.port,
+                        use_ssl: uri.scheme == 'https') { |http| response = http.head(uri.path.to_s) }
+        httpmodifytime = Time.parse(response['Last-Modified']).to_i
```

Comment:
Yeah, me neither. I just let the Puppet run fail now when the Last-Modified header is not in place. If someone wants support for extra headers, this can always be added later.